lib/logstash/inputs/elasticsearch.rb

```diff
@@ -563,12 +563,20 @@ def setup_basic_auth(user, password)
   end
 
   def setup_api_key(api_key)
-    return {} unless (api_key && api_key.value)
+    return {} unless (api_key&.value)
 
-    token = ::Base64.strict_encode64(api_key.value)
+    token = is_base64?(api_key.value) ?  api_key.value : Base64.strict_encode64(api_key.value)
     { 'Authorization' => "ApiKey #{token}" }
   end
 
+  def is_base64?(string)
+    begin
+      string == Base64.strict_encode64(Base64.strict_decode64(string))
+    rescue ArgumentError
+      false
+    end
+  end
+
```

Comment:
https://github.com/rubocop/ruby-style-guide?tab=readme-ov-file#implicit-begin

```suggestion
  def is_base64?(string)
    string == Base64.strict_encode64(Base64.strict_decode64(string))
  rescue ArgumentError
    false
  end

```

Reply:
Done!